proposals/0173-Read-Generic-Network-Signal-data.md

```diff
@@ -16,277 +16,306 @@ To enable OEM proprietary mobile apps to read raw CAN/Network data by specifying
-      <description>See Result</description>
-      <element name="SUCCESS"/>
-      <element name="INVALID_DATA"/>
-      <element name="OUT_OF_MEMORY"/>
-      <element name="TOO_MANY_PENDING_REQUESTS"/>
-      <element name="APPLICATION_NOT_REGISTERED"/>
-      <element name="GENERIC_ERROR"/>
-      <element name="REJECTED"/>
-      <element name="VEHICLE_DATA_NOT_AVAILABLE"/>
-      <element name="USER_DISALLOWED"/>
-    </param>
+* As you see, there are 3 sub items, “ambientLightSensorStatus”, “highBeamsOn” and “lowBeamsOn”. Each of these individual data types and references
+* You many notice that “ambientLightSensorStatus” has data type as “AmbientLightStatus”. This enumeration is read from existing Mobile API.
+  * This can be extrapolated to all existing data types. For new data types enumerations, we’d still either need to add to Mobile API or create a new structure for vehicle data enumerations.
+
+### Next example touches on a *new* OEM specific vehicle data item, _engineState_
+
+Sample value for map:
+```json
+{
+	"id":"engineState",
+	"dataType":"Struct",
+	"reference":"OEM_REF_STRUCT",
+	"engineCoolantTemp" : {
+		"dataType":"Integer",
+		"reference":"OEM_REF_ENG_COOL_TEMP",
+		"array":false,
+		"mandatory":false
+	},
+	"engineO2Sat":{
+		"dataType":"Float",
+		"reference":"OEM_REF_ENG_O2_SAT",				
+		"mandatory":false
+	},
+	"engineState":{
+		"dataType":"String",
+		"reference":"OEM_REF_ENG_STATE"
+	},
+	"engineServiceRequired":{
+		"dataType":"Boolean",
+		"reference":"OEM_REF_ENG_SER_REQ",
+		"array":false,
+		"mandatory":false
+	}
+}
+```
+Sample response from module:
+```json
+{
+	"engineState": {
+		"engineCoolantTemp":140,
+		"engineO2Sat":95.5,
+		"engineState":"NORMAL",
+		"engineServiceRequired":false
+	}
+}
+```
+#### engineState would ideally be an enum if it was a Standardized data type. But since this it OEM example of vehicle data on the fly, we can use String data type.
+
+* _engineState_ would need to be added to both SDL core and proxy schema. Once new schema is downloaded by both core and proxy, an app may request _engineState_ vehicle data item.
+* It just shows the capability to add a new vehicle data item which OEM can utilize to update the file at cloud. SDL can get updates on this file along with PT update and thus would be able to process new vehicle data
+* Again, each sub items have individual data type and references. So as long as OEM updates the reference file, Mobile apps can access the new data.
+
+### Also, this approach can also be used to read entire 8 bytes CAN message itself
+
+Sample value for map:
+```json
+{
+	"id":"messageName",
+	"dataType":"String",
+	"reference":"OEM_REF_MSG",
+	"array":false,
+	"mandatory":true
+}
+```
+Sample response from module:
+```json
+{
+	"messageName": "AB 04 D1 9E 84 5C B8 22"
+}
+```
+* Data type is String so that it can accommodate a byte string for CAN message. OEM_REF_MSG points to a message definition in OEM mapping table.
```

Comment:
Change to:
**The Data type is the string** so that it can accommodate a byte string for CAN message. OEM_REF_MSG points to a message definition in **the** OEM mapping table.

Reply:
Update this please